### 1. 创建一个长度为10的一维全为0的ndarray对象，然后让第5个元素等于1

In [ ]:
import numpy as np
a = np.zeros(10)
print(a)
a[4] = 1
print(a)

### 2. 请分别按一下两种方式各创建一个以10起始，以49结束的等差序列ndarray对象
2.1 包含20个元素  
2.2 每隔3取值一个元素

In [7]:
a = np.linspace(10,49,20)
print(a)
#linspace:起始值，终止值(包含)，数目，等差序列
b = np.arange(10,50,3)
print(b)

[10.         12.05263158 14.10526316 16.15789474 18.21052632 20.26315789
 22.31578947 24.36842105 26.42105263 28.47368421 30.52631579 32.57894737
 34.63157895 36.68421053 38.73684211 40.78947368 42.84210526 44.89473684
 46.94736842 49.        ]
[10 13 16 19 22 25 28 31 34 37 40 43 46 49]


### 3. 将2中最后一个ndarray翻转

In [10]:
c = b[::-1]
#[开始：结束：步长]
print(c)

[49 46 43 40 37 34 31 28 25 22 19 16 13 10]


### 4. 使用np.random.random创建一个10*10的ndarray对象，并打印出最大最小元素

### 5. 创建一个10*10的ndarray对象，且矩阵边界全为1，里面全为0

### 6. 创建一个每一行都是从0到4的5*5矩阵

### 7. 创建一个长度为10的随机数组并将最大值替换为0

### 8. 生成长度为10的随机数组m
8.1 运行print(t.argsort())  
8.2 运行t.sort()  
比较两个排序命令的区别

### 9. 给定一个4维矩阵，如何得到最后两维的和？
可使用np.random.randint(0,10,(3,3,3,3))生成0到9范围内的随机整数矩阵

### 10. 给定数组[1, 2, 3, 4, 5]，如何得到在这个数组的每个元素之间插入3个0后的新数组？

### 11. 给定一个二维矩阵，如何交换其中两行的元素？

### 12. 创建一个5 \* 3 随机矩阵和一个3 \* 2随机矩阵，求矩阵积

如果你不知道什么是矩阵积，请看这里：https://baike.baidu.com/item/%E7%9F%A9%E9%98%B5%E4%B9%98%E6%B3%95/5446029?fr=aladdin

### 13. 生成一个5 \*5 的0到10之间的随机整数矩阵，矩阵的每一行的元素都减去该行的平均值

### 14. 生成如下矩阵：
[[0 1 0 1 0 1 0 1]  
 [1 0 1 0 1 0 1 0]  
 [0 1 0 1 0 1 0 1]  
 [1 0 1 0 1 0 1 0]  
 [0 1 0 1 0 1 0 1]  
 [1 0 1 0 1 0 1 0]  
 [0 1 0 1 0 1 0 1]  
 [1 0 1 0 1 0 1 0]]  
 14.1 要求使用np.zeros创建8*8的矩阵  
 14.2 np.fromfunction创建

### 15、正则化一个5\*5随机矩阵  
【正则的概念：假设a是矩阵中的一个元素，max/min分别是矩阵元素的最大最小值，则正则化后a = (a - min)/(max - min)】